kymatio/scattering1d/frontend/torch_frontend.py

```diff
@@ -67,41 +67,28 @@ def load_filters(self):
                     n += 1
 
     def scattering(self, x):
+        self.load_filters()
         ScatteringBase1D._check_runtime_args(self)
         ScatteringBase1D._check_input(self, x)
-
-        batch_shape = x.shape[:-1]
-        signal_shape = x.shape[-1:]
-
-        x = x.reshape((-1, 1) + signal_shape)
-
-        self.load_filters()
+        x_shape = self.backend.shape(x)
+        batch_shape, signal_shape = x_shape[:-1], x_shape[-1:]
+        x = self.backend.reshape_input(x, signal_shape, n_inserted_dims=1)
 
         S = scattering1d(x, self.backend, self.psi1_f, self.psi2_f, self.phi_f,\
                          max_order=self.max_order, average=self.average, pad_left=self.pad_left, pad_right=self.pad_right,
                         ind_start=self.ind_start, ind_end=self.ind_end, oversampling=self.oversampling)
 
-        if self.out_type == 'array':
-            S = self.backend.concatenate([x['coef'] for x in S])
-            scattering_shape = S.shape[-2:]
-            new_shape = batch_shape + scattering_shape
-            S = S.reshape(new_shape)
-        elif self.out_type == 'dict':
-            S = {x['n']: x['coef'] for x in S}
-            for k, v in S.items():
-                # NOTE: Have to get the shape for each one since we may have
-                # average == False.
-                scattering_shape = v.shape[-2:]
-                new_shape = batch_shape + scattering_shape
-                S[k] = v.reshape(new_shape)
-        elif self.out_type == 'list':
-            for x in S:
-                x.pop('n')
-                scattering_shape = x['coef'].shape[-1:]
-                new_shape = batch_shape + scattering_shape
-                x['coef'] = x['coef'].reshape(new_shape)
-
-        return S
+        n_kept_dims = 1 + (self.out_type=="dict")
+        for n, path in enumerate(S):
+            S[n]['coef'] = self.backend.reshape_output(
+                path['coef'], batch_shape, n_kept_dims=n_kept_dims)
+
+        if self.out_type=='array':
```

Comment:
same here `self.out_type == 'array'`

Reply:
182ae3b

(search-and-replaced 12 times)